### ISSUE TITLE
Request for new material : Boron Carbide

### ISSUE THREAD

**Issue by Zecolas** [mctools/ncrystal#207]
Needed in simulations of neutron shielding inside the scattering chamber.

**Comment by marquezj** [comment]
Hi @Zecolas. Boron carbide is a highly absorbing material, which means the most important thing is the absorption, not the structure for the scattering. Because of that, it can be modeled as an unstructured solid material:

`solid::B4C/2.52gcm3/B_is_0.95_B10_0.05_B11`

(you can change the density and isotopic composition as needed)

That is one of the examples when you browse the NCrystal library:

https://github.com/mctools/ncrystal-notebooks/blob/main/notebooks/ncrystal1_basic_01_Introduction_and_Python_API.ipynb

**Comment by tkittel** [comment]
@Zecolas very interesting. If you wish, we can easily create a proper crystalline B4C .ncmat file for you in that case?

**Comment by Zecolas** [comment]
Hi, @tkittel. Yes, I would love to have a crystalline B4C file!

**Comment by marquezj** [comment]
@Zecolas you can do it yourself too. Just follow the tutorial here:

https://github.com/mctools/ncrystal-notebooks/blob/main/notebooks/ncrystal2_advanced_01_Creating_materials_and_the_NCMATComposer.ipynb

https://github.com/mctools/ncrystal-notebooks/blob/main/notebooks/ncrystal2_advanced_02_Import_crystal_structure_from_CIF_or_databases.ipynb

and use the CIF file for B4C from the Materials Project:

https://next-gen.materialsproject.org/materials/mp-696746

This is the result with the default Debye temperature. 

![Image](https://github.com/user-attachments/assets/5cb964e4-a6f2-4b90-bcad-9d04e1e6977b)

**Comment by tkittel** [comment]
Yup, that might be a good exercise. And you can also just use the cmdline tool `ncrystal_cif2ncmat`.

However, as always the trick is to find a trustworthy CIF file as input. Here are my tries, comparing the CIF file above with two files from the crystallography open database (COD), showing just the bragg diffraction:

![Image](https://github.com/user-attachments/assets/eb5b5798-a8d5-43d3-9259-aa82a373030a)

For the one marked `_DTedited" I changed the debye temperature from 300K (default if nothing is known) to 900K, which was more or less what I saw in one of the COD entries.

Looking at the above, I feel like disqualifying the (very old) COD4124697, and going with one of the others. Since the other COD entry uses non-unit site occupancy factors, which as the moment is only partially supported in NCrystal, I would recommend using the material db entry with an increased debye temperature.

```
> ncrystal_cif2ncmat  --spacegroup "R -3 m:H"  mpid::696746 --debyetemp=900
Querying materialsproject.org for entry mp-696746
Retrieving MaterialsDoc documents: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 14074.85it/s]
Adding mp_696746.cif to local file cache in $NCRYSTAL_ONLINEDB_CACHEDIR
Attempting to load CIF data with gemmi
WARNING: Overriding spacegroup to "R -3 m:H" due to explicit request
Self-consistency of structure was verified by spglib
Verifying that resulting ncmat data can be loaded
Writing autogen_CB4_sg166_mp696746.ncmat
```

I attach it here for reference (gzipped since github wont allow the attachment otherwise).

[autogen_CB4_sg166_mp696746.ncmat.gz](https://github.com/user-attachments/files/19872362/autogen_CB4_sg166_mp696746.ncmat.gz)

**Comment by Zecolas** [comment]
Thank you @tkittel @marquezj !!! I tried as per your guidance and have got satisfying results. This is a very good exercise!!!

**Comment by tkittel** [comment]
Great @Zecolas. Feel free to post some plot here with some results, I am rather curious :-)

Did you use the ncmat file I uploaded? If you are happy with it, I might add it to the official NCrystal data library, since you are not the first to ask for B4C.

Also, I assume that you are aware that you can use the `atomdb` cfg parameter, if you wish to enrich the boron :-)
